c_src/workitems.cc

```diff
@@ -691,20 +874,20 @@ char* EncodeVarint64(char* dst, uint64_t v) {
  */
 work_result RangeScanTask::operator()()
```

Comment:
The second catch block calls
ErlUtil::formatBinary(unsigned char* buf, size_t size),  which does not
throw.  It just formats the passed buf to look like an erlang term, and
returns the formatted string.

(You're thinking of
ErlUtil::formatBinary(ErlNifEnv* env, ERL_NIF_TERM term),
which can throw.  And a throw during a catch would  just throw another
exception that needs to be caught)

So this is not an unprotected throw.



On Mon, Oct 19, 2015 at 3:23 PM, Matthew Von-Maszewski <
notifications@github.com> wrote:

> In c_src/workitems.cc
> <https://github.com/basho/eleveldb/pull/162#discussion_r42433985>:
>
> > @@ -691,20 +874,20 @@ char* EncodeVarint64(char* dst, uint64_t v) {
> >   */
> >  work_result RangeScanTask::operator()()
>
> Please see second catch block. github will not let me comment on line 1063
> of new code. ErlUtil::formatBinary() can send a throw. What happens to a
> throw that occurs during a catch? I don't know.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/basho/eleveldb/pull/162/files#r42433985>.
>